From: Exequiel Echevarria <exemetal@hotmail.com>
Subject: [PATCH 3/4] osdep: update cfg80211_rtw_remain_on_channel for Linux 7.2/7.3+

Linux 7.2 updated remain_on_channel to take an rx_addr parameter and return int.
Linux 7.3 updated the cookie argument from pointer to 64-bit integer value.

---
 os_dep/linux/ioctl_cfg80211.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
index c914e78..d481023 100644
--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -6399,7 +6399,11 @@ inline s32 rtw_cfg80211_get_last_ro_ch_passing_ms(_adapter *adapter)
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 2, 0))
+static int cfg80211_rtw_remain_on_channel(struct wiphy *wiphy,
+#else
 static s32 cfg80211_rtw_remain_on_channel(struct wiphy *wiphy,
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 6, 0))
 	struct wireless_dev *wdev,
 #else
@@ -6414,7 +6418,17 @@ static s32 cfg80211_rtw_remain_on_channel(struct wiphy *wiphy,
         unsigned int duration,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 3, 0))
+	u64 cookie
+#else
 	u64 *cookie
+#endif
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 2, 0))
+        , const u8 *rx_addr)
+#else
+        )
+#endif
 {
-- 
2.45.0
